mathics/core/expression.py

```diff
@@ -909,7 +907,7 @@ def rules():
         for rule in rules():
             result = rule.apply(new, evaluation, fully=False)
             if result is not None:
-                if isinstance(result, BoxConstruct):
```

Comment:
When this was added an alarm should have gone off - Expression needs knows that there is something called BoxConstruct? The fact that it needed to be imported in that funky way should have raised its own alarm. 

The code is trying to figure out whether we need to consider expanding an expression further, and there should be a way to do that without knowing anything more that this. 

There might be a better way to do this than it check `ExpandOnce` instead. And perhaps there is a better name for this as well.  Some version of "never try to expand",  or "result is final"?

Reply:
What about `FinalForm`?

Reply:
I don't have a strong feeling one way or another. Looking that the wolfram docs Form often refers to the way something is presented. However in the code we use Form as a way of matching structure. And I don't think that's quite right either. When Symbol "x" gets evaluated to 5 its shape stays the same, it is the _evaluation_ here that we are avoiding. So it si more like FinalEvaluation.

Reply:
`BoxConstruct` was more than just a flag over an `Expression` to avoid reevaluation: It was something that shares a part of its interface with `Expression`, but 1. stores information in a different way, and 2. has an extra interface to produce certain kind of outputs (like Graphics, Graphics3d, or another kind of still not implemented objects, like waveform sounds or manipulable expressions). This class has members of the form `boxes_to_*` that produces the representation of the Box structure.    What maybe makes sense is to tag (by adding a property or by inheritance) `BoxConstruct` as `ExpandOnce`